x/valset-pref/validator_set.go

```diff
@@ -207,6 +208,84 @@ func (k Keeper) PreformRedelegation(ctx sdk.Context, delegator sdk.AccAddress, e
 	return nil
 }
 
+// WithdrawDelegationRewards withdraws all the delegation rewards from the validator in the val-set.
+// Delegation reward is collected by the validator and in doing so, they can charge commission to the delegators.
+// Rewards are calculated per period, and is updated each time validator delegation changes. For ex: when a delegator
+// receives new delgation the rewards can be calculated by taking (total rewards before new delegation - the total current rewards).
+func (k Keeper) WithdrawDelegationRewards(ctx sdk.Context, delegatorAddr string) error {
+	delegator, err := sdk.AccAddressFromBech32(delegatorAddr)
+	if err != nil {
+		return err
+	}
+
+	// check if there is existing staking position that's not val-set
+	delegations := k.stakingKeeper.GetDelegatorDelegations(ctx, delegator, math.MaxUint16)
+
+	// get the existing validator set preference
+	existingSet, found := k.GetValidatorSetPreference(ctx, delegatorAddr)
```

Comment:
I guess I'm confused -- I thought the module design was always such that `GetValidatorSetPreference` returns your existing staking distribution if you haven't set one.

Is that not the case or only in query logic?

Reply:
Unfortunately that is not the case, `GetValidatorSetPreference` only returns val-set delegations

Reply:
ok, imo we need to change that (which I thought was the initial design for valset pref)

Reply:
for sure, it wont be a massive change tbh. Logic will be something like. 
```
func() GetValidatorSetPreference {
valsetDels, vsFound := getvalsetDels(...) 
 if !vsFound {
   existingstakingDels, stFound := getstakingDels(...) 
   if !stFound {
    return err ("No delegations") 
 }
return existingstakingDels 
} 
return valsetDels 
}
```

What about cases where we need both existing valset and staking position? Do you think this function should handle that too?

Reply:
1) if a valset is set return that. (ignore staking position, that may diverge)
2) You can derive the weight on the fly based on ratio of staked amounts

Reply:
I'm down to merge this PR, but lets do the above as a release blocking feature.

Can you make an issue? Then we can merge this PR

Reply:
added issue https://github.com/osmosis-labs/osmosis/issues/3848